### PR TITLE
issue #1790 Refactor DockerContainerOperationManagerTest

### PR DESCRIPTION
**Pull request by VolLol**
Rewrite integration tests for unit tests.

### CODE REVIEW

File: api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 package com.epam.pipeline.manager.docker;
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java:17]: Could you please update license?

Comment (ekazachkova) [reply api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java:17]: Could you please remove `AbstractManagerTest` inheritance?

File: api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java [hunk 2]
```diff
@@ -167,7 +179,8 @@ public void runShouldBeStillRunningIfCommandCannotExecute() throws IOException,
         when(sshConnection.waitFor(anyLong(), any())).thenReturn(true);
         when(sshConnection.exitValue()).thenReturn(CANNOT_EXECUTE_EXIT_CODE);
         final PipelineRun run = pipelineRun();
-
+        doReturn(null).when(preferenceManager).getPreference(SystemPreferences.BASE_API_HOST);
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java]: Why do we need this mock?

File: api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java [hunk 3]
```diff
@@ -109,21 +113,28 @@
...
+        final AwsRegion region = new AwsRegion();
         region.setRegionCode("eu-central-1");
         region.setId(1L);
         when(regionManager.load(any())).thenReturn(region);
         when(cloudFacade.startInstance(Mockito.anyLong(), anyString()))
                 .thenReturn(CloudInstanceOperationResult.builder().status(Status.ERROR)
                         .message(INSUFFICIENT_INSTANCE_CAPACITY).build());
+        doReturn(CloudInstanceState.STOPPED).when(cloudFacade).getInstanceState(anyLong());
 
         PipelineRun run = pipelineRun();
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java]: Could you add final to `run`?

File: api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java [hunk 4]
```diff
@@ -109,21 +113,28 @@
     @Mock
     private PipelineConfigurationManager pipelineConfigurationManager;
 
+    @Mock
+    private MessageHelper messageHelper;
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java:115]: Could you please fix `pmdTest` task? (add `@SuppressWarnings("PMD.UnusedPrivateField")` to field)

**Review by ekazachkova — CHANGES_REQUESTED** [review]

File: api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java (hunk 2 above)
Comment (VolLol) [reply api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java]: We use this mock in the `launchPauseRunScript` method in the following line 

`final String pauseRunCommand = DockerPauseCommand.builder()
                .runPauseScriptUrl(pauseRunScriptUrl)
                .api(preferenceManager.getPreference(SystemPreferences.BASE_API_HOST))`

Comment (ekazachkova) [reply api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java]: Right, but why do we need to mock this method? (`preferencemanager` returns `null` for each method excepting mocked)

File: api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java [hunk 5]
```diff
@@ -76,6 +78,7 @@
     private static final long REGION_ID = 1L;
     private static final Long RUN_ID = 2L;
     private static final int CANNOT_EXECUTE_EXIT_CODE = 126;
+    private static final int PAUSE_TIMEOUT = 86400;
 
     @InjectMocks
     @Spy
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java:83]: Do we still need `Autowired` here?

**Review by ekazachkova — CHANGES_REQUESTED** [review]

File: api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java (hunk 2 above)
Comment (VolLol) [reply api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java]: Yes, you're right. It's not necessary mock.

File: api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java (hunk 5 above)
Comment (VolLol) [reply api/src/test/java/com/epam/pipeline/manager/docker/DockerContainerOperationManagerTest.java:83]: You are absolutely right. This `Autowired` annotation not necessary in this class.

**Review by ekazachkova — APPROVED** [review]

**Pull request merged**